pkg/api/meta/restmapper.go

```diff
@@ -148,6 +148,23 @@ func kindToResource(kind string, mixedCase bool) (plural, singular string) {
 	return
 }
 
+// ResourceSingularlizer converts a resource name from plural to singular (e.g., from pods to pod)
+func ResourceSingularlizer(resource string) (singular string) {
```

Comment:
This probably needs to be a method on the interface RESTMapper, rather than a static global method.  The decision to undo pluralization belongs to the interface that pluralizes them.  During initialization of the rest mapper we can build the map of initial singular to plural.